contracts/protocol/libraries/logic/CalldataLogic.sol

```diff
@@ -0,0 +1,323 @@
+      deadline := and(shr(152, args), 0xFFFFFFFF)
+      permitV := and(shr(184, args), 0xFF)
+    }
+
+    return (asset, amount, interestRateMode, deadline, permitV);
+  }
+
+  /**
+   * @notice Decodes compressed swap borrow rate mode params to standard params
+   * @param reservesList The addresses of all the active reserves
+   * @param args The packed swap borrow rate mode params
+   * @return The address of the underlying reserve
+   * @return The interest rate mode, 1 for stable 2 for variable debt
+   */
+  function decodeSwapBorrowRateMode(mapping(uint256 => address) storage reservesList, bytes32 args)
+    internal
+    view
+    returns (address, uint256)
+  {
+    uint16 assetId;
+    uint256 interestRateMode;
+
+    assembly {
+      assetId := and(args, 0xFFFF)
+      interestRateMode := and(shr(16, args), 0xFF)
+    }
+
+    return (reservesList[assetId], interestRateMode);
+  }
+
+  /**
+   * @notice Decodes compressed rebalance stable borrow rate params to standard params
+   * @param reservesList The addresses of all the active reserves
+   * @param args The packed rabalance stable borrow rate params
+   * @return The address of the underlying reserve
+   * @return The address of the user to rebalance
+   */
+  function decodeRebalanceStableBorrowRate(
+    mapping(uint256 => address) storage reservesList,
+    bytes32 args
+  ) internal view returns (address, address) {
+    uint16 assetId;
+    address user;
+    assembly {
+      assetId := and(args, 0xFFFF)
+      user := and(shr(16, args), 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
+    }
+    return (reservesList[assetId], user);
+  }
+
+  /**
+   * @notice Decodes compressed set user use reserve as collateral params to standard params
+   * @param reservesList The addresses of all the active reserves
+   * @param args The packed set user use reserve as collateral params
+   * @return The address of the underlying reserve
+   * @return True if to set using as collateral, false otherwise
+   */
+  function decodeSetUserUseReserveAsCollateral(
+    mapping(uint256 => address) storage reservesList,
+    bytes32 args
+  ) internal view returns (address, bool) {
+    uint16 assetId;
+    bool useAsCollateral;
+    assembly {
+      assetId := and(args, 0xFFFF)
+      useAsCollateral := and(shr(16, args), 0x1)
+    }
+    return (reservesList[assetId], useAsCollateral);
+  }
+
+  /**
+   * @notice Decodes compressed liquidation call params to standard params
+   * @param reservesList The addresses of all the active reserves
+   * @param args1 The first half of packed liquidation call params
+   * @param args2 The second half of the packed liquidation call params
+   * @return The address of the underlying collateral asset
+   * @return The address of the underlying debt asset
+   * @return The address of the user to liquidate
+   * @return The amount of debt to cover
+   * @return True if receiving aTokens, false otherwise
+   */
+  function decodeLiquidationCall(
```

Comment:
we used the `Params` suffix for supply/borrow/repay/withdraw but it wasnt used for the rest of the functions, i would either remove in all of them or keep it in all of them for consistency